llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1182,8 +1187,13 @@ PassBuilder::buildModuleSimplificationPipeline(OptimizationLevel Level,
   const bool IsCtxProfUse =
       !UseCtxProfile.empty() && Phase == ThinOrFullLTOPhase::ThinLTOPreLink;
 
+  // Enable cold function coverage instrumentation if
+  // InstrumentColdFuncOnlyPath is provided.
+  const bool IsColdFuncOnlyInstrGen = PGOInstrumentColdFunctionOnly =
```

Comment:
Setting this global variable here has introduced a data race detected by TSan when different threads set up different pass pipelines. Was this intentional? Did you mean to use `==` instead of `=`?

Reply:
Thanks for pointing out. It's intentional to set the global variable, I wasn't aware it will cause data race. I will try to change it to local one.

Reply:
I'm sorry but had to revert it in 06e28ed84f3f0a05b493699965e18e7a5aeaccd2 and d924a9ba03a05b417676e84f6c81aac44f907f71

Reply:
Should be fixed by https://github.com/llvm/llvm-project/pull/114364